streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -152,23 +152,20 @@ public Subscription subscription(Set<String> topics) {
      * @param topicToTaskIds Map that contains the topic names to be created
      * @param compactTopic If true, the topic should be a compacted topic. This is used for
      *                     change log topics usually.
-     * @param outPartitionInfo If true, compute and return all partitions created
      * @param postPartitionPhase If true, the computation for calculating the number of partitions
      *                           is slightly different. Set to true after the initial topic-to-partition
      *                           assignment.
      * @return
      */
     private Map<TopicPartition, PartitionInfo> prepareTopic(Map<String, Set<TaskId>> topicToTaskIds,
                                                             boolean compactTopic,
-                                                            boolean outPartitionInfo,
                                                             boolean postPartitionPhase) {
         Map<TopicPartition, PartitionInfo> partitionInfos = new HashMap<>();
-        // if ZK is specified, prepare the internal source topic before calling partition grouper
-        if (internalTopicManager != null) {
-            log.debug("Starting to validate internal topics in partition assignor.");
-
-            for (Map.Entry<String, Set<TaskId>> entry : topicToTaskIds.entrySet()) {
-                String topic = entry.getKey();
+        log.debug("Starting to validate internal topics in partition assignor.");
+        for (Map.Entry<String, Set<TaskId>> entry : topicToTaskIds.entrySet()) {
+            String topic = entry.getKey();
+            // if ZK is specified, prepare the internal source topic before calling partition grouper
+            if (internalTopicManager != null) {
```

Comment:
Instead of moving this check internally to the loop, can we add an `else` call after the original if condition that, if `topicToTaskIds` is not empty, log a single warning entry that `Topics {} do not exist but ...` with `topicToTaskIds.keySet().toList()`. This will not swamp multiple warning log entries.